### PR TITLE
chore: Code Simplification and Optimization

### PR DESCRIPTION
**Pull request by sky-coderay**
Description of changes:
1. Simplification of obtaining _code_hash_: Instead of using the _has_code_ variable and the _then()_ method, the _code_hash_ is directly computed through the _if !account.code.is_empty()_ condition, which makes the code more straightforward.
2. Simplification of the loop over the storage: Instead of using _iter().filter().try_for_each()_, I used a regular _for_ loop, making the code easier to understand.
3. Bytecode handling: The redundant variable has been removed, and _code_hash_ is directly written to the database if it exists.

These changes make the code more readable and understandable, while functionality remains unchanged.

### CODE REVIEW

**Review by mattsse — COMMENTED** [review]
lgtm, only have 1 question about changed test

File: testing/ef-tests/src/models.rs [hunk 1]
```diff
@@ -165,22 +165,25 @@ impl State {
...
-                )?;
-                tx.put::<tables::HashedStorages>(
-                    hashed_address,
-                    StorageEntry { key: keccak256(storage_key), value: *v },
-                )
-            })?;
-        }
 
+            for (k, v) in &account.storage {
+                if !v.is_zero() {
```
[the remark this answers is not in the document]
Comment (onbjerg) [reply testing/ef-tests/src/models.rs:174]: could be even easier to read if we did `if v.is_zero() { continue }` to remove a level of indentation

**Comment by sky-coderay** [comment]
@mattsse Hi! Just a reminder about my PR . If everything looks good, it would be great if you could merge it. Thanks! 😊

**Review by mattsse — APPROVED** [review]
thanks, undid the unrelated test change that was blocking this pr

**Pull request merged**